src/common/config_parser.cpp

```diff
@@ -154,6 +154,34 @@ void ConfigParser::addOptionsModel(cli::CLIWrapper& cli) {
   cli.add<bool>("--tied-embeddings-all",
       "Tie all embedding layers and output layer");
 
+  // S2S multi-head and multi-hop attention
+  cli.add<int> ("--dec-attention-heads",
+      "Number of parallel attention heads in s2s decoder",
+      1);
+  cli.add<int>("--dec-attention-hops",
+      "Number of sequential attention hops in s2s decoder",
+      1);
+  cli.add<int>("--dec-attention-lookup-dim",
+      "Vector size of the s2s attention lookup mechanism, per head per hop (-1 same the encoder state dimension)",
+      -1);
+  cli.add<int>("--dec-attention-projection-dim",
+      "Project the context state to this size, per hop (-1 disables projection)",
+      -1);
+  cli.add<bool>("--dec-attention-independent-heads",
+      "Use completely independent heads in s2s multi-head attention. "
+        "Automatically enabled by --dec-attention-bilinear-lookup)");
+  cli.add<bool>("--dec-attention-bilinear-lookup",
+      "Use log-bilinear softmax attention in s2s decoder (currently disabled due to a bug)");
```

Comment:
A reminder for ourselves that we shouldn't merge the PR until we take a look at this bug.